wren-ai-service/src/web/v1/services/ask.py

```diff
@@ -233,25 +278,31 @@ async def ask(
                         invalid_generation_results=failed_dry_run_results,
                         project_id=ask_request.project_id,
                     )
-                    valid_generation_results += sql_correction_results["post_process"][
-                        "valid_generation_results"
-                    ]
+                    if valid_generation_results := sql_correction_results[
+                        "post_process"
+                    ]["valid_generation_results"]:
+                        valid_sql_summary_results = (
+                            await self._add_summary_to_sql_candidates(
+                                valid_generation_results,
+                                ask_request.query,
+                                ask_request.configurations.language,
+                            )
+                        )
+                        api_results += [
+                            AskResult(**result) for result in valid_sql_summary_results
+                        ]
```

Comment:
the reference is already pointed to the attribute of response object, thus add new results into this variable, it might be occurring more than three candidates back. i think we need to check how many candidates had before putting them into the array.

Reply:
Here is the example to show the case

```python
from dataclasses import dataclass


@dataclass
class Demo:
    id: str


@dataclass
class Resp:
    demos: list[Demo]


results = []

results += [Demo(id="1")]

resp = Resp(demos=results)

print(resp)


results += [Demo(id="2")]

print(resp)

```